openedx/core/djangoapps/import_export/courses/views.py

```diff
@@ -0,0 +1,526 @@
+            self._save_request_status(request, courselike_string, -1)
+            if course_dir.isdir():  # pylint: disable=no-value-for-parameter
+                shutil.rmtree(course_dir)
+                log.info(
+                    "Course import %s: Temp data cleared", courselike_key
+                )
+
+            log.exception("error importing course")
+            return JsonResponse(
+                {
+                    'error_message': str(exception),
+                    'stage': -1
+                },
+                status=400
+            )
+
+        # try-finally block for proper clean up after receiving last chunk.
+        try:
+            # This was the last chunk.
+            log.info("Course import %s: Upload complete", courselike_key)
+            self._save_request_status(request, courselike_string, 1)
+
+            tar_file = tarfile.open(temp_filepath)
+            try:
+                safetar_extractall(
+                    tar_file,
+                    (course_dir + '/').encode('utf-8'))
+            except SuspiciousOperation as exc:
+                self._save_request_status(request, courselike_string, -1)
+                return JsonResponse(
+                    {
+                        'error_message': 'Unsafe tar file. Aborting import.',
+                        'suspicious_operation_message': exc.args[0],
+                        'stage': -1
+                    },
+                    status=400
+                )
+            finally:
+                tar_file.close()
+
+            log.info(
+                "Course import %s: Uploaded file extracted", courselike_key
+            )
+            self._save_request_status(request, courselike_string, 2)
+
+            # find the 'course.xml' file
+            def get_all_files(directory):
+                """
+                For each file in the directory, yield a 2-tuple of (file-name,
+                directory-path)
+                """
+                for dirpath, _dirnames, filenames in os.walk(directory):
+                    for filename in filenames:
+                        yield (filename, dirpath)
+
+            def get_dir_for_fname(directory, filename):
+                """
+                Returns the dirpath for the first file found in the directory
+                with the given name.  If there is no file in the directory with
+                the specified name, return None.
+                """
+                for fname, dirpath in get_all_files(directory):
+                    if fname == filename:
+                        return dirpath
+                return None
+
+            dirpath = get_dir_for_fname(course_dir, root_name)
+            if not dirpath:
+                self._save_request_status(request, courselike_string, -2)
+                return JsonResponse(
+                    {
+
+                        'error_message': _(
+                            'Could not find the {0} file in the'
```

Comment:
This works in Python, but not javascript (in JS, the "gettext" must be on the same line as the string). However, I think splitting the string here doesn't make sense - it's not a long string, and you're going to end up with "thepackage" as one word. Just put the whole string together on one line.

Also for strings that are translated, please don't use positional variables. Named variables give a richer context. See http://edx.readthedocs.org/projects/edx-developer-guide/en/latest/internationalization/i18n.html#style-guidelines for guidelines.